packages/block/src/block.ts

```diff
@@ -33,6 +42,7 @@ export class Block {
   public readonly header: BlockHeader
   public readonly transactions: TypedTransaction[] = []
   public readonly uncleHeaders: BlockHeader[] = []
+  public readonly withdrawals?: Withdrawal[]
```

Comment:
Thinking about this for some time now: `withdrawals` have this type `Withdrawal` here with all these non-deterministic types like `BigIntLike`:

```typescript
export type Withdrawal = {
  index: BigIntLike
  validatorIndex: BigIntLike
  address: AddressLike
  amount: BigIntLike
}
```

While this always comes in handy for input data I wonder if this is an optimal design choice for the internal representation of this data. In the case of withdrawals, this `Withdrawal` type atm actually serves for *both* puroposes.

With this kind of design we just can't be sure what type these internal withdrawal parameters like `index` or `address` really is. My fear is that this will lead to a myriad of follow-up problems. For test cases, client usages,... one can never really count on what the type really is here and either cast here, make code comments, add complex procedural choice conditional switches for values on processing, or minimally: one always need to convert with `withdrawalToBufferArray()`.

I would have a relatively strong tendency that we should switch here to deterministic values, I guess that will safe us from some substantial amount of follow-up trouble.

So I would suggest to actually have *two* types here, the one from above renamed to `WithdrawalData` or similar and used for withdrawal inputs.

And another one `Withdrawal` with deterministic types. Following our tendency to represent internal values not always with Buffer any more but with more "matching" types this would be something like the following I guess:


```typescript
export type Withdrawal = {
  index: BigInt
  validatorIndex: BigInt
  address: Address
  amount: BigInt
}
```

I know that this is some extra effort to rewrite now with various code already in place but I am relatively sure that this more than pay off in the future.

Reply:
Update: Haha https://github.com/ethereumjs/ethereumjs-monorepo/pull/2401/files#diff-6ef8827503f0272ab6d926170261d659f841b39a00e354abcd69ea5635c8eae1

Just discovered on the follow up PR from @g11tech.

At least we think in impressively similar lines. 🤩 😂